libexec/pyenv-init

```diff
@@ -91,21 +91,40 @@ function help_() {
     ;;
   esac
 
-  { echo "# Add pyenv executable to PATH by adding"
-    echo "# the following to ~/.profile:"
+  { echo '# Add pyenv executable to PATH and'
+    echo '# enable shims by adding the following'
+    case "$shell" in
+    zsh )
+      echo '# to ~/.profile and ~/.zprofile:'
+      ;;
+    * )
+      echo '# to ~/.profile:'
+      ;;
+    esac
     echo
     case "$shell" in
     fish )
       echo 'set -Ux PYENV_ROOT $HOME/.pyenv'
       echo 'set -Ux fish_user_paths $PYENV_ROOT/bin $fish_user_paths'
+      echo 'pyenv init --path | source'
       ;;
     * )
       echo 'export PYENV_ROOT="$HOME/.pyenv"'
       echo 'export PATH="$PYENV_ROOT/bin:$PATH"'
+      echo 'eval "$(pyenv init --path)"'
```

Comment:
Thank you for grouping these lines together! Seeing two sections that both need to go in `.profile` was extremely confusing.

Reply:
In fact, adding `shims` to `PATH` and the function to the shell have been presented as an optional step to offer a "bare minimal impact" installation option to help with Pyenv adoption -- so that prospective users aren't scared off by high upfront requirements and the fear that it would break their systems.

But now with `~/.profile` shenanigans added, two steps that involve `~/.profile` instead of one have made the instruction with the optional step too complicated.

Does the above still hold? Maybe it's time to stop worrying about that and lump all the parts together if practical?